### PR TITLE
Add script to autogenerate Terraform module docs.

### PR DESCRIPTION
**Pull request by morgante**
This generates a simple index of modules based on the module registry (with capacity for filtering or excluding modules). In the future, we should set up a GitHub action to continuously regenerate this.

### CODE REVIEW

File: docs/meta/generate-index.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+# limitations under the License.
+
+import os
+import sys
+import argparse
+import yaml
+
+import requests
+from collections import OrderedDict
+from jinja2 import Environment, FileSystemLoader
```
Comment (bharathkkb) [inline docs/meta/generate-index.py:24]: Shouldn't we have a requirements file?

File: docs/meta/generate-index.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+  for module in data.get("modules", []):
+    item = root.add_child_data(module)
+    id = module.get("id")
+    r = requests.get(f"{TERRAFORM_REGISTRY_BASE}/modules/{id}")
+    data = r.json()
+    for child in data.get("submodules", []):
+      item.add_child_data(child)
+
+def render_index(index, templates_dir, docs_dir):
```
Comment (bharathkkb) [inline docs/meta/generate-index.py:94]: looks like we are not using `templates_dir`?

File: docs/meta/generate-index.py [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+  def description(self):
+    return self.data.get("description")
+
+  def add_child_data(self, data):
+    child = self.add_child(IndexItem(data))
+    child.data = {**child.data, **data}
+    return child
+
+  def add_child(self, child):
```
[the remark this answers is not in the document]
Comment (morgante) [reply docs/meta/generate-index.py:55]: They're just subdirectories so I don't think there's an easy way to add them. But we could probably adopt a convention (ex. first line of README) if we wanted to.

File: docs/meta/generate-index.py [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+    f.truncate()
+    yaml.dump(root, f)
+
+  render_index(root, meta_dir, docs_dir)
+
+def argparser():
+  parser = argparse.ArgumentParser(description='Generate index of blueprints')
+  parser.add_argument('docs_dir', metavar='F')
+
+  parser.add_argument('--skip-refresh', default=False, action='store_true')
```
[the remark this answers is not in the document]
Comment (morgante) [reply docs/meta/generate-index.py:141]: This is autocreated by argparser.

File: docs/meta/generate-index.py [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+    rendered = template.render(modules=modules)
+
+    with open(os.path.join(docs_dir, output_file), "w") as f:
+      f.write(rendered)
+
+def main(argv):
+  parser = argparser()
+  args = parser.parse_args(argv[1:])
+  docs_dir = args.docs_dir
+  meta_dir = os.path.join(docs_dir, "meta")
```
[the remark this answers is not in the document]
Comment (morgante) [reply docs/meta/generate-index.py:119]: You can run it from anywhere, it accepts the docs directory path as an argument.

**Pull request merged**